subsys/elfloader/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+if(CONFIG_DYNAMIC_MODULE)
```

Comment:
as you are excluding everything in this file then I think it's better to not load the file in the first place, see other comment.

```suggestion
```